prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -538,32 +539,44 @@ def create_buses(lines):
     return buses
 
 
-def create_transformers(bus):
+def create_transformers(bus, lines, transformer_designs):
     """Add transformers between buses within the same substation. The assumed topology
     is that the highest-voltage bus in each substation is connected via a tansformer to
     every other voltage.
 
-    :param pandas.DataFrame bus: columns 'sub_id' and 'baseKV'
+    :param pandas.DataFrame bus: columns 'sub_id' and 'baseKV'.
+    :param pandas.DataFrame lines: columns 'from_bus_id', 'to_bus_id', and 'rateA'.
+    :param pandas.DataFrame transformer_designs: representative transformer data. Index
+        is (low_kV, high_kV), columns are 'x', 'r', and 'MVA'. 'x' and 'r' values are
+        per-unit, 'MVA' is in megawatts.
     :return: (*pandas.DataFrame*) -- each row is one transformer, columns are
-        ["from_bus_id", "to_bus_id"].
+        ["from_bus_id", "to_bus_id", "x", "r", and "rateA"].
     """
     bus_pairs = [
         (b, volt_series.sort_values().index[i + 1])
         for sub, volt_series in bus.groupby("sub_id")["baseKV"]
         for i, b in enumerate(volt_series.sort_values().index[:-1])
         if len(volt_series) > 1
     ]
+    bus_pairs_df = pd.DataFrame(bus_pairs, columns=["from_bus_id", "to_bus_id"])
+    transformer_details = bus_pairs_df.apply(
+        lambda x: estimate_transformers(x, lines, bus["baseKV"], transformer_designs),
+        axis=1,
+    )
+    single_transformers = pd.concat([bus_pairs_df, transformer_details], axis=1)
+    repeated_transformers = single_transformers.drop(columns="number").reindex(
+        single_transformers.index.repeat(single_transformers["number"])
+    )
 
-    return pd.DataFrame(bus_pairs, columns=["from_bus_id", "to_bus_id"])
+    return repeated_transformers.reset_index(drop=True)
 
 
-def add_impedance_and_rating(branch, bus_voltages, line_overrides=None):
+def add_lines_impedances_ratings(branch, line_overrides={}):
```

Comment:
Why switch default to `{}`?

Reply:
Oops, I think that got messed up during a rebase. Fixed.